static/app/components/modals/createTeamModal.tsx

```diff
@@ -1,52 +1,47 @@
-import {Component, Fragment} from 'react';
+import {Fragment} from 'react';
 
 import {ModalRenderProps} from 'app/actionCreators/modal';
 import {createTeam} from 'app/actionCreators/teams';
 import {Client} from 'app/api';
 import CreateTeamForm from 'app/components/teams/createTeamForm';
 import {t} from 'app/locale';
 import {Organization, Team} from 'app/types';
-import withApi from 'app/utils/withApi';
+import useApi from 'app/utils/useApi';
 
-type Props = {
+type Props = ModalRenderProps & {
   api: Client;
   organization: Organization;
   onClose?: (team: Team) => void;
-} & ModalRenderProps;
-
-class CreateTeamModal extends Component<Props> {
-  handleSubmit = (data: {slug: string}, onSuccess: Function, onError: Function) => {
-    const {organization, api} = this.props;
-    createTeam(api, data, {orgId: organization.slug})
-      .then((resp: Team) => {
-        this.handleSuccess(resp);
-        onSuccess(resp);
-      })
-      .catch((err: Error) => {
-        onError(err);
-      });
-  };
-
-  handleSuccess(team: Team) {
-    if (this.props.onClose) {
-      this.props.onClose(team);
+};
+
+function CreateTeamModal({Body, Header, ...props}: Props) {
+  const {onClose, closeModal, organization} = props;
```

Comment:
Should these be destructured in one place?

Reply:
We pass down the `props`, which is why I did this.